tests/cpptests/test_cpp_iterator_intersection.cpp

```diff
@@ -0,0 +1,353 @@
+/*
+ * Copyright Redis Ltd. 2016 - present
+ * Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ * the Server Side Public License v1 (SSPLv1).
+ */
+
+#include "gtest/gtest.h"
+#include <map>
+
+#include "rmutil/alloc.h"
+#include "iterator_util.h"
+
+#include "src/iterators/intersection_iterator.h"
+#include "src/iterators/inverted_index_iterator.h"
+#include "src/inverted_index.h"
+#include "src/forward_index.h"
+
+class IntersectionIteratorCommonTest : public ::testing::TestWithParam<std::tuple<unsigned, std::vector<t_docId>>> {
+protected:
+    std::vector<std::vector<t_docId>> docIds;
+    std::vector<t_docId> resultSet;
+    QueryIterator *ii_base;
+
+    void SetUp() override {
+        unsigned numChildren;
+        std::tie(numChildren, resultSet) = GetParam();
+        // Set docIds so the intersection of all children is resultSet.
+        // Make sure that some ids are unique to each child
+        docIds.resize(numChildren);
+        t_docId id = 1;
+        for (auto &childIds : docIds) {
+            // Copy the resultSet to each child as a base
+            childIds = resultSet;
+            // Add some unique ids to each child. Mock constructors will ensure that the ids are unique and sorted.
+            for (size_t i = 0; i < 100; i++) {
+                childIds.push_back(id++);
+            }
+        }
+        // Create children iterators
+        auto children = (QueryIterator **)rm_malloc(sizeof(QueryIterator *) * numChildren);
+        for (unsigned i = 0; i < numChildren; i++) {
```

Comment:
Somehow it would be nice to "randomize" the original order of the Children iterator to validate the sorting in constructor